Native-Build missing Reflection-Config for BedrockCohereEmbeddingResponse

If built with Quarkus Native-Build we get the following error while trying to call `EmbeddingStoreContentRetriever.retrieve()` and therefore can't retrieve the Embeddings:
```
com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `dev.langchain4j.model.bedrock.BedrockCohereEmbeddingResponse`: cannot deserialize from Object value (no delegate- or property-based Creator): this appears to be a native image, in which case you may need to configure reflection for the class that is to be deserialized
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2]
	at dev.langchain4j.model.bedrock.Json.fromJson(Json.java:31)
	at dev.langchain4j.model.bedrock.BedrockCohereEmbeddingModel.embedAll(BedrockCohereEmbeddingModel.java:79)
	at dev.langchain4j.model.embedding.EmbeddingModel.embed(EmbeddingModel.java:36)
	at dev.langchain4j.model.embedding.EmbeddingModel.embed(EmbeddingModel.java:26)
	at dev.langchain4j.model.embedding.EmbeddingStoreProducer_ProducerMethod_queryEmbeddingModel_P7lTBJKDrG88wfmBd9A1-f27E-o_ClientProxy.embed(Unknown Source)
	at dev.langchain4j.rag.content.retriever.EmbeddingStoreContentRetriever.retrieve(EmbeddingStoreContentRetriever.java:227)
	at ***.IncidentMemoryRetrievalAugmentor.lambda$gracefulDegradation$0(IncidentMemoryRetrievalAugmentor.java:83)
	at dev.langchain4j.rag.DefaultRetrievalAugmentor.process(DefaultRetrievalAugmentor.java:168)
	at dev.langchain4j.rag.DefaultRetrievalAugmentor.augment(DefaultRetrievalAugmentor.java:150)
	at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.doImplement0(AiServiceMethodImplementationSupport.java:279)
	at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.doImplement(AiServiceMethodImplementationSupport.java:203)
	at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport.implement(AiServiceMethodImplementationSupport.java:178)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$1.apply(MethodImplementationSupportProducer.java:31)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$1.apply(MethodImplementationSupportProducer.java:28)
	at io.quarkiverse.langchain4j.runtime.aiservice.SpanWrapper.wrap(SpanWrapper.java:32)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$2.apply(MethodImplementationSupportProducer.java:40)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$2.apply(MethodImplementationSupportProducer.java:37)
	at io.quarkiverse.langchain4j.runtime.aiservice.MetricsCountedWrapper.wrap(MetricsCountedWrapper.java:22)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$2.apply(MethodImplementationSupportProducer.java:40)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$2.apply(MethodImplementationSupportProducer.java:37)
	at io.quarkiverse.langchain4j.runtime.aiservice.MetricsTimedWrapper$2.get(MetricsTimedWrapper.java:42)
	at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:69)
	at io.quarkiverse.langchain4j.runtime.aiservice.MetricsTimedWrapper.wrap(MetricsTimedWrapper.java:39)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$2.apply(MethodImplementationSupportProducer.java:40)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1$2.apply(MethodImplementationSupportProducer.java:37)
	at io.quarkiverse.langchain4j.runtime.aiservice.MethodImplementationSupportProducer$1.implement(MethodImplementationSupportProducer.java:46)
	at ***
	at ***
	at ***
	at io.smallrye.context.impl.wrappers.SlowContextualSupplier.get(SlowContextualSupplier.java:21)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromItemSupplier.subscribe(UniCreateFromItemSupplier.java:28)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:35)
	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.lambda$subscribe$0(UniRunSubscribeOn.java:27)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$15.runWith(VertxCoreRecorder.java:677)
	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
	at java.base@25.0.3/java.lang.Thread.run(Thread.java:1474)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `dev.langchain4j.model.bedrock.BedrockCohereEmbeddingResponse`: cannot deserialize from Object value (no delegate- or property-based Creator): this appears to be a native image, in which case you may need to configure reflection for the class that is to be deserialized
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2]
	at com.fasterxml.jackson.databind.DeserializationContext.reportBadDefinition(DeserializationContext.java:1935)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:415)
	at com.fasterxml.jackson.databind.DeserializationContext.handleMissingInstantiator(DeserializationContext.java:1417)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1511)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:340)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:5052)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3954)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3922)
	at dev.langchain4j.model.bedrock.Json.fromJson(Json.java:29)
	... 44 more
```

(`IncidentMemoryRetrievalAugmentor.lambda$gracefulDegradation` is just a wrapper around retrieve to log errors and so on).

We added the following as a workaround in our Application, but i'd have expected this extension to ensure it works in native build
```java
@RegisterForReflection(classNames = "dev.langchain4j.model.bedrock.BedrockCohereEmbeddingResponse")
public class ReflectionConfiguration {
}
```

### Versions / Models
**Model-Provider:** AWS Bedrock
**Embedding-Model:** eu.cohere.embed-v4:0
**Embedding-Store:** MongoDB Vector Search
**Quarkus Version:** 3.33.2
**Quarkiverse-Langchain4J-Bedrock Version:** 1.7.5 (through quarkus-langchain4j-bom)
**LangChain4J Version:** 1.11.7-beta19 (through quarkus-langchain4j-bom)

--- a/model-providers/bedrock/deployment/src/main/java/io/quarkiverse/langchain4j/bedrock/deployment/BedrockProcessor.java
+++ b/model-providers/bedrock/deployment/src/main/java/io/quarkiverse/langchain4j/bedrock/deployment/BedrockProcessor.java
@@ -33,6 +33,7 @@ import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
 import io.quarkus.proxy.ProxyConfigurationRegistry;
 import io.quarkus.vertx.core.deployment.CoreVertxBuildItem;
@@ -60,9 +61,12 @@ public class BedrockProcessor {
     }
 
     @BuildStep
-    void nativeSupport(BuildProducer<ServiceProviderBuildItem> serviceProviderProducer) {
+    void nativeSupport(BuildProducer<ServiceProviderBuildItem> serviceProviderProducer,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassProducer) {
         serviceProviderProducer
                 .produce(ServiceProviderBuildItem.allProvidersFromClassPath(ConfigSourceInterceptor.class.getName()));
+        reflectiveClassProducer.produce(
+                ReflectiveClassBuildItem.builder("dev.langchain4j.model.bedrock.BedrockCohereEmbeddingResponse").build());
     }
 
     @BuildStep

